caffe2/contrib/tensorrt/tensorrt_tranformer.cc

```diff
@@ -0,0 +1,423 @@
+      LOG(WARNING) << "Cannot get shape of " << s;
+    } else {
+      auto* tensor_type = value_info.mutable_type()->mutable_tensor_type();
+      tensor_type->set_elem_type(
+          ::ONNX_NAMESPACE::TensorProto_DataType::TensorProto_DataType_FLOAT);
+      auto* shape = tensor_type->mutable_shape();
+      for (int i = 0 ; i < it->second.dims().size(); ++i) {
+        auto* dim = shape->add_dim();
+        dim->set_dim_value(it->second.dims(i));
+      }
+    }
+  }
+  return r;
+}
+
+void PruneWeights(const NetDef& pred_net, NetDef* init_net) {
+  std::unordered_set<std::string> used_weights;
+  for (const auto& op: pred_net.op()) {
+    for (const auto& i: op.input()) {
+      used_weights.emplace(i);
+    }
+  }
+
+  int last = init_net->op_size();
+  for (int i = 0; i < last;) {
+    if (!used_weights.count(init_net->op(i).output(0))) {
+      if (i != last - 1) {
+        init_net->mutable_op()->SwapElements(i, last - 1);
+      } else {
+        ++i;
+      }
+      --last;
+    } else {
+      ++i;
+    }
+  }
+
+  if (last < init_net->op_size()) {
+    init_net->mutable_op()->DeleteSubrange(last, init_net->op_size() - last);
+  }
+
+  LOG(INFO) << "New init_net op size: " << init_net->op_size();
+}
+
+void FillModelInfo(::ONNX_NAMESPACE::ModelProto* model) {
+  model->set_ir_version(::ONNX_NAMESPACE::Version::IR_VERSION);
+  model->set_producer_name("caffe2");
+  auto* opset_id = model->add_opset_import();
+  opset_id->set_domain("");
+  opset_id->set_version(3);
+}
+} // namespace
+
+OperatorDef TensorRTTransformer::BuildTrtOp(
+    const std::string& onnx_model_str,
+    const std::unordered_map<std::string, std::vector<int>>&
+        output_size_hints) {
+  OperatorDef op;
+  op.set_type("TensorRT");
+
+  TrtLogger logger;
+  auto trt_builder = InferObject(nvinfer1::createInferBuilder(logger));
+  auto trt_network = InferObject(trt_builder->createNetwork());
+  auto importer = InferObject(onnx2trt::createImporter(trt_network.get()));
+  auto status = importer->import(onnx_model_str.data(), onnx_model_str.size(), false);
+  if (status.is_error()) {
+    CAFFE_THROW(MakeString(
+        "TensorRTTransformer ERROR: ",
+        status.file(),
+        ":",
+        status.line(),
+        " In function ",
+        status.func(),
+        ":\n",
+        "[",
+        status.code(),
+        "] ",
+        status.desc()));
+  }
+  trt_builder->setMaxBatchSize(max_batch_size_);
+  trt_builder->setMaxWorkspaceSize(max_workspace_size_);
+  trt_builder->setDebugSync(debug_builder_);
+  auto trt_engine =
+      InferObject(trt_builder->buildCudaEngine(*trt_network.get()));
+
+  // Set up inputs/outputs
+  int num_bindings = trt_engine->getNbBindings();
+  for (int b = 0; b < num_bindings; ++b) {
+    const auto& name = trt_engine->getBindingName(b);
+    if (trt_engine->bindingIsInput(b)) {
+      op.add_input(name);
+    } else {
+      op.add_output(name);
+    }
+  }
+
+  auto engine_plan = InferObject(trt_engine->serialize());
+
```

Comment:
`caffe2/utils/proto_utils.h` may be helpful here

Reply:
@Maratyszcza  I actually met with a weird linkage error when I tried to use `MakeArgument` here. It complains  that `MakeArugument` symbol is missing during runtime. Any idea?